File: CheckStatus.java
public class CheckStatus {
static int SIZE = 5;
static int[] queue = new int[SIZE];
static int frontA = -1, rearA = -1;

static boolean isEmptyArray() { return (frontA == -1 || frontA > rearA); }
static boolean isFullArray() { return (rearA == SIZE - 1); }

//for linked lists
static class Node {
    int data; Node next;
    Node(int val) { data = val; }
}
static Node frontL = null, rearL = null;

static boolean isEmptyLinkedList() { return (frontL == null); }

public static void main(String[] args) {
    System.out.println("Array Queue Empty? " + (isEmptyArray() ? "Yes" : "No"));
    rearA = SIZE - 1;
    System.out.println("Array Queue Full? " + (isFullArray() ? "Yes" : "No"));
    System.out.println("Linked List Queue Empty? " + (isEmptyLinkedList() ? "Yes" : "No"));
}
}

File: CheckStatus.c
#include <stdio.h>
#include <stdlib.h>

#define SIZE 5
int queue[SIZE];
int frontA = -1, rearA = -1;

int isEmptyArray() { return (frontA == -1 || frontA > rearA); }
int isFullArray() { return (rearA == SIZE - 1); }

//for linked lists
struct Node {
int data;
struct Node* next;
};
struct Node* frontL = NULL;
struct Node* rearL = NULL;

int isEmptyLinkedList() { return (frontL == NULL); }

int main() {
printf("Array Queue Empty? %s\n", isEmptyArray() ? "Yes" : "No");
rearA = SIZE - 1;
printf("Array Queue Full? %s\n", isFullArray() ? "Yes" : "No");

printf("Linked List Queue Empty? %s\n", isEmptyLinkedList() ? "Yes" : "No");
return 0;
}
